nautilus_trader/data/engine.pyx

```diff
@@ -1646,34 +1622,26 @@ cdef class DataEngine(Component):
                 f"data[-1].ts_init={data[-1].ts_init}, {ts_now=}",
             )
 
-        # Special handling for instruments requests to ensure proper filtering
-        if isinstance(request, RequestInstruments) and request.start is not None:
```

Comment:
Correct to remove this post processing for now. We can bring it back later based on a parameter the user can control.